### ISSUE TITLE
[Local Statistics] API Integration

### ISSUE THREAD

**Issue by ferndot** [joshua-s/coronavirus-diary#179]
Requirements:
- Case statistics for the specified location should be fetched from an API exposed by CovidNearMe.org.
  - This API will require the user’s zip code or country and return case statistics with county-level or country-level granularity.
  - The API accepts two character ISO 3166-2 alpha 2 country codes, with Kosovo represented as XK.

------

# Locations API

Endpoint: http://covidnearme.org/api/v1/locations
Query parameters:
* `zip`: 5-digit zip code
* `country`: ISO 3166-1 alpha-2 country code
* Either zip or country must be specified

## Response schema
Note: the figures here are totals cases/deaths/recoveries, not new cases/deaths/recoveries.

```
[
  {
    "date": string (YYYY-MM-DD),
    "cases": int,
    "deaths": int,
    "recoveries": int,
    “name”: string
  },
  {
    "date": string (YYYY-MM-DD),
    "cases": int,
    "deaths": int,
    "recoveries": int,
    “name”: string
  },
  {
    "date": string (YYYY-MM-DD),
    "cases": int,
    "deaths": int,
    "recoveries": int,
    “name”: string
  }
]
```

An invalid zip will result in the following response:
```
{
   "message": "No Results for Specified Zip; Possibly Invalid"
}
```

If the requested zip code or country has no results, the API will return an empty list.

Deaths, cases, and recoveries are for the county or country in which the zip code is located. Data starts at 3/23 because that’s when county-level data became available again.

**Comment by clocksmith** [comment]
@joshua-s Is there already an API that exists? Where can it be tested? What does the response look like?

**Comment by ferndot** [comment]
@clocksmith just received the specs. Will be adding them to the issue over the weekend!

**Comment by ferndot** [comment]
Updated!

**Comment by clocksmith** [comment]
Thanks!

The API works for some zip codes (https://covidnearme.org/api/v1/locations?zip=33326), but is missing data for other zip codes. For example, these zip codes in Brooklyn and Manhattan in NY are showing 0 cases:
https://covidnearme.org/api/v1/locations?zip=11211 (Brooklyn)
https://covidnearme.org/api/v1/locations?zip=10011 (Manhattan)

Is the data in the API still being populated?

**Comment by ferndot** [comment]
@clocksmith the data in the API is ultimately sourced from the Johns Hopkins dataset. I will surface these missing zip codes and let you know why they are empty.

**Comment by ferndot** [comment]
Will be fixed by 1195402d203b5274dadfb5c6b34c526615d30117
